Data/HashMap/Base.hs

```diff
@@ -623,6 +624,17 @@ lookupCont absent present !h0 !k0 !m0 = go h0 k0 0 m0
         | otherwise = absent (# #)
 {-# INLINE lookupCont #-}
 
+-- | /O(log n)/ Return the value to which the specified key is mapped,
+-- or 'Nothing' if this map contains no mapping for the key.
```

Comment:
We should probably revisit the `O(log n)` claim at some point, since our hashing is not too random for, e.g., `Int`. But that doesn't have to be handled in this PR.